### DISCUSSION TITLE
Condition coverage?

### DISCUSSION THREAD

**Discussion by gvl2023** [linux-test-project/lcov#326] [Q&A] [answered]
Looks like gcov can produce condition-coverage information (-fcondition-coverage, gcov --conditions), can it be shown in the lcov html report?

**Comment by henry2cox** [comment]
Current upstreamed version:  no/not yet.
But probably fairly soon.

**Comment by henry2cox** [comment]
see sha 8f4d7839abee

Note that this is tested only with gcc/14.2.0.
You can find an example in the 'example' directory (though this one is kind of bogus because all the expressions are trivial, so MC/DC and branch coverage are identical).

**Comment by henry2cox** [comment, marked as answer]
FWIW:  TOT adds support for LLVM - tested with v18.1.3.
Please see the `genhtml` man page for a discussion of how MC/DC interpretation differs between gcc and llvm.
Testcase `.../tests/lcov/mcdc` illustrates some of those differences.
